index/scorch/persister.go

```diff
@@ -266,16 +266,20 @@ func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot) error {
 		for k, v := range s.root.internal {
 			newIndexSnapshot.internal[k] = v
 		}
-		for _, filename := range filenames {
-			delete(s.ineligibleForRemoval, filename)
-		}
+
 		rootPrev := s.root
 		s.root = newIndexSnapshot
 		s.rootLock.Unlock()
 		if rootPrev != nil {
 			_ = rootPrev.DecRef()
 		}
 	}
+	// unlock the files for clean up
```

Comment:
can we not use the phrase "unlock" since lock has very specific meaning, perhaps say

allow files to become eligible for removal again